scipy/stats/stats.py

```diff
@@ -1229,12 +1229,11 @@ def skew(a, axis=0, bias=True, nan_policy='propagate'):
 
     m2 = moment(a, 2, axis)
     m3 = moment(a, 3, axis)
-    zero = (m2 == 0)
-    vals = _lazywhere(~zero, (m2, m3),
-                      lambda m2, m3: m3 / m2**1.5,
-                      0.)
+    with np.errstate(all='ignore'):
+        zero = (np.abs(m2) <= (np.finfo(m2.dtype).resolution * a.mean(axis))**2)
```

Comment:
`m2` is the result of `moment(a, 2, axis)`, which should never produce a negative value, so we could drop the use of `abs` here:
```suggestion
        zero = (m2 <= (np.finfo(m2.dtype).resolution * a.mean(axis))**2)
```